bundles/gcp/src/main/java/org/apache/gravitino/gcs/fs/GCSFileSystemProvider.java

```diff
@@ -43,9 +45,18 @@ public FileSystem getFileSystem(Path path, Map<String, String> config) throws IO
     Configuration configuration = new Configuration();
     FileSystemUtils.toHadoopConfigMap(config, GRAVITINO_KEY_TO_GCS_HADOOP_KEY)
         .forEach(configuration::set);
+
+    if (enableCredentialProvidedByGravitino(config)) {
+      configuration.set(GCS_TOKEN_PROVIDER_IMPL, GCSCredentialsProvider.class.getName());
+    }
+
     return FileSystem.newInstance(path.toUri(), configuration);
   }
 
+  private boolean enableCredentialProvidedByGravitino(Map<String, String> config) {
+    return null != config.get(GravitinoFileSystemCredentialProvider.GVFS_CREDENTIAL_PROVIDER);
+  }
```

Comment:
I think we can also move this code to the common place.

Reply:
done